packages/opentelemetry-tracing/src/Span.ts

```diff
@@ -178,6 +182,34 @@ export class Span implements api.Span, ReadableSpan {
     return true;
   }
 
+  recordException(exception: api.Exception, time: api.TimeInput = hrTime()) {
+    const attributes: api.Attributes = {};
+    if (!exception) {
+      return;
+    }
+    if (typeof exception === 'string') {
+      attributes[ExceptionAttribute.MESSAGE] = exception;
+    } else {
+      if (exception.name) {
+        attributes[ExceptionAttribute.TYPE] = exception.name;
+      }
+      if (exception.message) {
+        attributes[ExceptionAttribute.MESSAGE] = exception.message;
+      }
+      if (exception.stack) {
+        attributes[ExceptionAttribute.STACKTRACE] = exception.stack;
+      }
+    }
+
+    // these are minimum requirements from spec
+    if (
+      attributes[ExceptionAttribute.TYPE] ||
+      attributes[ExceptionAttribute.MESSAGE]
+    ) {
+      this.addEvent(ExceptionEventName, attributes as api.Attributes, time);
+    }
```

Comment:
else log? If you send something that passes the if-check, but get no event, you might be confused.

Should we consider adding the event anyway, even if the "required" attributes are missing? _some_ information is likely better than no information. We could set `type` to "UNKNOWN_ERROR" in that case.

Reply:
the spec says the type or message is minimum

Reply:
with regards to logging, should the log be done as an error or debug only ?

Reply:
I would say `warn` level probably

Reply:
> Should we consider adding the event anyway, even if the "required" attributes are missing? _some_ information is likely better than no information. We could set `type` to "UNKNOWN_ERROR" in that case.

The spec doesn't say anything about it, I would rather ask in spec what they think about it, if we should log something or not. Do you know what other sigs are doing in such case ?

Reply:
I would prefer to log a warning instead of having a uncompliant event, specially since exporters might ignore them anyway

Reply:
I would prefer to log a warning instead of having a uncompliant event, specially since exporters might ignore them anyway

Reply:
If those fields are mandatory, I'd prefer an error to be thrown in development time instead of leaving developers to guess what happened on the swallowed exceptions (logs may be verbose in cases, it may not easy to notice one single log).

Reply:
We don't throw an error - this is general rule in whole api/sdk. We can just log that something is wrong and this is what is happening in this case.

Reply:
@obecny is right here. One of the core principles is that we don't throw errors, even in cases where the lib is used wrong.